core/node/node_framework/src/implementations/layers/web3_api/server/bridge_addresses.rs

```diff
@@ -0,0 +1,47 @@
+use std::time::Duration;
+
+use zksync_node_api_server::web3::state::BridgeAddressesHandle;
+use zksync_web3_decl::{
+    client::{DynClient, L2},
+    namespaces::ZksNamespaceClient,
+};
+
+use crate::{StopReceiver, Task, TaskId};
+
+#[derive(Debug)]
+pub struct BridgeAddressesUpdaterTask {
+    pub bridge_address_updater: BridgeAddressesHandle,
+    pub main_node_client: Box<DynClient<L2>>,
+    pub update_interval: Option<Duration>,
+}
+
+#[async_trait::async_trait]
+impl Task for BridgeAddressesUpdaterTask {
+    fn id(&self) -> TaskId {
+        "bridge_addresses_updater_task".into()
+    }
+
+    async fn run(self: Box<Self>, mut stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        let update_interval = self.update_interval.unwrap_or(Duration::from_secs(30));
+
+        while !*stop_receiver.0.borrow_and_update() {
+            match self.main_node_client.get_bridge_contracts().await {
```

Comment:
Nit: Since calling the main node could take a large time, there *may* make sense to use `tokio::time::interval` here. OTOH, it may put additional load to an already struggling server, so maybe it's better to retain the current approach.

Reply:
IMO it doesn't matter, in practice bridge contracts will only be changed once and it's not a big deal if they will reach ENs not in 30 seconds but 35